### ISSUE TITLE
ConnectionError when building DAL microbit samples

### ISSUE THREAD

**Issue by bcopy** [carlosperate/docker-microbit-toolchain#2]
Hi,
thanks for this image. When following instructions, the ```yotta build``` fails with a connection error.

Here are the steps I follow ;
```bash
$ git clone https://github.com/lancaster-university/microbit-samples
$ docker run -v $(pwd):/home --rm ghcr.io/carlosperate/microbit-toolchain:latest yotta build
```

result :
```bash
Unable to find image 'ghcr.io/carlosperate/microbit-toolchain:latest' locally
latest: Pulling from carlosperate/microbit-toolchain
171857c49d0f: Pull complete 
419640447d26: Pull complete 
61e52f862619: Pull complete 
1274120bb463: Pull complete 
dc74b5100461: Pull complete 
Digest: sha256:82695ec17bccf34dd7636a62c232903af23aad47cd0fdbffbe8bf603132aaa4f
Status: Downloaded newer image for ghcr.io/carlosperate/microbit-toolchain:latest
info: get versions for bbc-microbit-classic-gcc
info: download bbc-microbit-classic-gcc@0.2.3 from the public module registry
error: 'ConnectionError' object has no attribute 'message'
```

Any idea ?

**Comment by carlosperate** [comment]
Hi @bcopy,

Yes, this is likely due to the Yotta deprecation, as the Yotta repositories have been taken out offline recently.
However, I thought that the micro:bit projects have been updated to fetch all dependencies via GitHub instead of the Yotta registry, I'll investigate.

**Comment by carlosperate** [comment]
Okay, so the problem is that this file has not been updated: https://github.com/lancaster-university/microbit/blob/master/.yotta.json

If you run this command before the `yotta build` it should work, I'll raise a PR for the `microbit-samples` repository:

```
docker run -v $(pwd):/home --rm ghcr.io/carlosperate/microbit-toolchain:latest yotta target bbc-microbit-classic-gcc-nosd@https://github.com/lancaster-university/yotta-target-bbc-microbit-classic-gcc-nosd
```

So:

```
$ git clone https://github.com/lancaster-university/microbit-samples
$ cd microbit-samples
$ docker run -v $(pwd):/home --rm ghcr.io/carlosperate/microbit-toolchain:latest yotta target bbc-microbit-classic-gcc-nosd@https://github.com/lancaster-university/yotta-target-bbc-microbit-classic-gcc-nosd
$ docker run -v $(pwd):/home --rm ghcr.io/carlosperate/microbit-toolchain:latest yotta build
```

**Comment by bcopy** [comment]
Indeed, thanks for the workaround !

**Comment by carlosperate** [comment]
Glad you got it working!

One small thing, in my workaround I configured the "No SoftDevice" version, which is the target without the Bluetooth stack. Should still work for a lot of examples, but will fail for anything trying to use the SoftDevice. The right command would be:

```
docker run -v $(pwd):/home --rm ghcr.io/carlosperate/microbit-toolchain:latest yotta target bbc-microbit-classic-gcc@https://github.com/lancaster-university/yotta-target-bbc-microbit-classic-gcc
```

I've raised the PR, so once that is merged the example should be working out-of-the-box again: https://github.com/lancaster-university/microbit-samples/pull/36
